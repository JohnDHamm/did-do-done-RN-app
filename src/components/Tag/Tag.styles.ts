import styled from 'styled-components/native';

export const Container = styled.View<
  Theme & { color: string; selected: boolean }
>`
  border-width: 1px;
  border-style: solid;
  border-color: ${(props) =>
    props.selected ? props.color : props.theme.unselectedTag};
  background-color: ${(props) =>
    props.selected ? props.color : 'transparent'};
  height: 24px;
  border-radius: 12px;
  padding: 0 10px;
`;

export const Label = styled.Text<Theme & { selected: boolean }>`
  font-size: 16px;
  color: ${(props) =>
    props.selected ? props.theme.tagText : props.theme.unselectedTag};
`;
