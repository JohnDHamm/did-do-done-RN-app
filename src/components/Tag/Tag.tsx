import React from 'react';
import { Container, Label } from './Tag.styles';

interface TagProps {
  label: string;
  color: string;
  selected: boolean;
}

const Tag: React.FC<TagProps> = ({ label, color, selected }) => {
  return (
    <Container color={color} selected={selected}>
      <Label selected={selected}>{label}</Label>
    </Container>
  );
};

export default Tag;
