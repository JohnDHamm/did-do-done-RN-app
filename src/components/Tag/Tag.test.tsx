import React from 'react';
import { cleanup } from 'react-native-testing-library';
import Tag from './Tag';
import { renderWithLightTheme } from '../../testing/helpers';

describe('Tag', () => {
  afterEach(cleanup);

  it('renders correctly as a selected tag', () => {
    const testRenderer = renderWithLightTheme(
      <Tag label="test" color="navy" selected={true} />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders correctly as an unselected tag', () => {
    const testRenderer = renderWithLightTheme(
      <Tag label="test" color="navy" selected={false} />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders the label prop', () => {
    const { getByText } = renderWithLightTheme(
      <Tag label="Tag 1" color="red" selected={false} />
    );

    expect(getByText('Tag 1')).not.toBeNull();
  });
});
