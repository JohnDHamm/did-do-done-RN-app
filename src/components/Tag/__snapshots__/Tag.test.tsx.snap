// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly as a selected tag 1`] = `
<View
  color="navy"
  selected={true}
  style={
    Array [
      Object {
        "backgroundColor": "navy",
        "borderColor": "navy",
        "borderRadius": 12,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 24,
        "paddingBottom": 0,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    selected={true}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 16,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Tag renders correctly as an unselected tag 1`] = `
<View
  color="navy"
  selected={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#c3c3c3",
        "borderRadius": 12,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 24,
        "paddingBottom": 0,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    selected={false}
    style={
      Array [
        Object {
          "color": "#c3c3c3",
          "fontSize": 16,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
